impacket/examples/ntlmrelayx/servers/rpcrelayserver.py

```diff
@@ -0,0 +1,428 @@
+#!/usr/bin/env python
+# Impacket - Collection of Python classes for working with network protocols.
+#
+# Copyright Fortra, LLC and its affiliated companies
+#
+# All rights reserved.
+#
+# This software is provided under a slightly modified version
+# of the Apache Software License. See the accompanying LICENSE file
+# for more information
+#
+# RPC Relay Server inspired from DCERPCServer
+#
+# Authors:
+#  Sylvain Heiniger / Compass Security (@sploutchy / https://www.compass-security.com)
+#
+# Description:
+#             This is the RPC server which relays the connections
+# to other protocols
+
+import socketserver
+import struct
+from impacket.dcerpc.v5.epm import *
+from impacket.dcerpc.v5.rpcrt import *
+from impacket.dcerpc.v5.dcomrt import *
+from impacket.ntlm import NTLMSSP_AUTH_NEGOTIATE, NTLMSSP_AUTH_CHALLENGE_RESPONSE, NTLMSSP_AUTH_CHALLENGE
+from impacket.smbserver import outputToJohnFormat, writeJohnOutputToFile
+from impacket.nt_errors import ERROR_MESSAGES, STATUS_SUCCESS
+from impacket.examples.ntlmrelayx.utils.targetsutils import TargetsProcessor
+from impacket.examples.ntlmrelayx.servers.socksserver import activeConnections
+
+
+class RPCRelayServer(Thread):
+    class RPCSocketServer(socketserver.ThreadingMixIn, socketserver.TCPServer):
+        def __init__(self, server_address, RequestHandlerClass, config):
+            self.config = config
+            self.daemon_threads = True
+            if self.config.ipv6:
+                self.address_family = socket.AF_INET6
+            socketserver.TCPServer.allow_reuse_address = True
+            socketserver.TCPServer.__init__(self, server_address, RequestHandlerClass)
+
+    class RPCHandler(socketserver.BaseRequestHandler):
+        def __init__(self, request, client_address, server):
+            self.client = None
+            self.target = None
+            self.auth_user = None
+            self.transport = None
+            self.request_header = None
+            self.request_pdu_data = None
+            self.request_sec_trailer = None
+            self.challengeMessage = None
+            socketserver.BaseRequestHandler.__init__(self, request, client_address, server)
+
+        def setup(self):
+            self.transport = DCERPCServer(self.request)
+            IObjectExporterCallBacks = {
+                5: self.send_ServerAlive2Response,
+            }
+            self.transport.addCallbacks(bin_to_uuidtup(IID_IObjectExporter), "135", IObjectExporterCallBacks)
+
+            IEPMCallBacks = {
+                3: self.handle_epmap
+            }
+            self.transport.addCallbacks(bin_to_uuidtup(MSRPC_UUID_PORTMAP), "135", IEPMCallBacks)
+
+            if self.server.config.target is None:
+                # Reflection mode, defaults to SMB at the target, for now
+                self.server.config.target = TargetsProcessor(singleTarget='SMB://%s:445/' % self.clientAddress[0])
```

Comment:
Fixing `client_address` property

```suggestion
                self.server.config.target = TargetsProcessor(singleTarget='SMB://%s:445/' % self.client_address[0])
```